src/app/needs/next-actions/components/NextActionsDisplay.tsx

```diff
@@ -0,0 +1,212 @@
+      a.priority!.order - b.priority!.order
+    );
+
+    const groupsMap: Record<number, { priority: { order: number; name: string}; needs: NeedDocument[] }> = {};
+    for (const need of sortedNeeds) {
+      const order = need.priority!.order;
+      if (!groupsMap[order]) {
+        groupsMap[order] = {
+          priority: need.priority!,
+          needs: []
+        };
+      }
+      groupsMap[order].needs.push(need);
+    }
+
+    return Object.values(groupsMap);
+  }, [highlightedNeeds]);
+
+  // Helper to get next actions for a given need
+  const getActionsForNeed = (needId: string) => {
+    return relatedNextActions.filter(action => action.need === needId);
+  };
+
+  // onToggleAction logic
+  const onToggleAction = async (action: NextActionDocument) => {
+    const highlighted = new Date(action.selectedExpiry) > new Date(action.timestamp);
+    const collectionName = "next_actions";
+
+    if (highlighted) {
+      // Un-highlight action:
+      // Remove last selectedTimestamp
+      const updatedTimestamps = [...action.selectedTimestamps];
+      updatedTimestamps.pop();
+      
+      // Set selectedExpiry back to the original timestamp
+      await database.updateDocument(
+        collectionName,
+        action.id,
+        "selectedTimestamps",
+        updatedTimestamps
+      );
+
+      await database.updateDocument(
+        collectionName,
+        action.id,
+        "selectedExpiry",
+        action.timestamp
+      );
+    } else {
+      // Highlight action:
+      // Add new selectedTimestamp
+      const updatedTimestamps = [...action.selectedTimestamps, new Date().toISOString()];
+
+      // Find the parent need to copy its selectedExpiry
+      const parentNeed = highlightedNeeds.find((need) => need.id === action.need);
+      if (!parentNeed) {
+        console.error("Parent need not found for action:", action);
+        return;
+      }
+
+      await database.updateDocument(
+        collectionName,
+        action.id,
+        "selectedTimestamps",
+        updatedTimestamps
+      );
+
+      await database.updateDocument(
+        collectionName,
+        action.id,
+        "selectedExpiry",
+        parentNeed.selectedExpiry
+      );
+    }
+
+    // After toggling, increment chainEnd to re-fetch and update UI
+    setChainEnd(prev => prev + 1);
+  };
+
+  const saveAndExit = () => {
+    // All changes are saved on toggle, so just navigate back
+    router.push("/needs");
+  };
+
+  return (
+    <>
+      <div className="w-11/12 m-auto">
+        {priorityGroups.length === 0 ? (
+          <p className="mb-5">
+            No highlighted needs found. Highlight some needs first, then come back here.
+          </p>
+        ) : (
+          priorityGroups.map((group, i) => (
+            <div key={i} className="mb-6">
+              <h3 className="text-xl font-bold mb-2">
+                {toTitleCase(group.needs[0].mood ?? "No mood")}: {toTitleCase(group.priority.name)}
```

Comment:
I don't think we need to be explicit about the mood, I believe it was meant a bit more subconsciously using the colours but "users don't like being told what mood they are in" was the feedback. Also this page focuses on the priority matrix and the figma just shows the order and priority (ie 1. do it now: / 2. schedule it: ...)
![image](https://github.com/user-attachments/assets/86ec7671-fdb8-4c8d-83df-2421ea6b974c)

Reply:
Fair, I'll change that